src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java

```diff
@@ -281,5 +291,83 @@ private void updateRequestStatus(AssignmentRequest assignmentRequest, Status sta
 
     }
 
+    @SuppressWarnings("unchecked")
+    public ResponseEntity<Void> deleteRoleAssignmentByQuery(MultipleQueryRequest multipleQueryRequest) {
+
+        //1. create the request Object
+        if (CollectionUtils.isNotEmpty(multipleQueryRequest.getQueryRequests())) {
+            request = parseRequestService.prepareDeleteRequest("", "", "", "");
+            request.setLog(DELETE_BY_QUERY);
+            assignmentRequest = new AssignmentRequest(request, Collections.emptyList());
+        } else {
+            throw new BadRequestException(V1.Error.BAD_REQUEST_MISSING_PARAMETERS);
+
+        }
+
+        //2. Call persistence service to store only the request
+        persistInitialRequestForDelete();
+
+        //3. retrieve all assignment records by multiple query
+        List<Assignment> requestedRoles = fetchRoleAssignmentsByMultipleQuery(multipleQueryRequest);
+        var responseHeaders = new HttpHeaders();
+        responseHeaders.add(
+            "Total-Records",
+            Long.toString(persistenceService.getTotalRecords())
+        );
+
+        if (requestedRoles.isEmpty()) {
+            requestEntity.setStatus(Status.APPROVED.toString());
```

Comment:
we should add the log that "No records found for deletion". This will help to identify which request couldn't find any records.

Reply:
Done.